node/src/main/scala/coop/rchain/node/NodeRuntime.scala

```diff
@@ -597,14 +604,20 @@ object NodeRuntime {
     def close(): F[Unit]
   }
 
-  def cleanup[F[_]: Sync: Log](runtime: Runtime[F], casperRuntime: Runtime[F]): Cleanup[F] =
+  def cleanup[F[_]: Sync: Log](
+      runtime: Runtime[F],
+      casperRuntime: Runtime[F],
+      deployStorageCleanup: F[Unit]
+  ): Cleanup[F] =
     new Cleanup[F] {
       override def close(): F[Unit] =
         for {
           _ <- Log[F].info("Shutting down interpreter runtime ...")
           _ <- runtime.close()
           _ <- Log[F].info("Shutting down Casper runtime ...")
           _ <- casperRuntime.close()
+          _ <- Log[F].info("Shutting down deploy storage ...")
+          _ <- deployStorageCleanup
```

Comment:
Is it makes sense to transform this to an applicative? Maybe closing can be done in parallel.

Reply:
In general, this piece of code should only operate on [Resources](https://typelevel.org/cats-effect/datatypes/resource.html). I have been planning to refactor this place for ages.